samples/socket-api/sample_test_run.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2019 Intel Corporation.  All rights reserved.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+
+import argparse
+import shlex
+import subprocess
+import sys
+import time
+import traceback
+
+WAMRC_CMD = "../../wamr-compiler/build/wamrc"
+
+def start_server(cmd, cwd):
+    app_server = subprocess.Popen(shlex.split(cmd), cwd=cwd)
+    return app_server
+
+def run_cmd(cmd, cwd):
+    qry_prc = subprocess.run(
+        shlex.split(cmd), cwd=cwd, check=False, capture_output=True
+    )
+    if (qry_prc.returncode != 0):
+        print("Run {} failed, return {}".format(cmd), qry_prc.returncode)
+        return
+    print("return code: {}, output:\n{}".format(qry_prc.returncode,
+                                                 qry_prc.stdout.decode()))
+
+def main():
+    """
+    GO!GO!!GO!!!
+    """
+    parser = argparse.ArgumentParser(description="run the sample and examine outputs")
+    parser.add_argument("working_directory", type=str)
+    parser.add_argument("--aot", action='store_true', help="Test with AOT")
+    args = parser.parse_args()
+
+    test_aot = False
+    suffix = ".wasm"
+    if not args.aot:
+        print("Test with interpreter mode")
+    else:
+        print("Test with AOT mode")
+        test_aot = True
+        suffix = ".aot"
+        wasm_files = [ "addr_resolve", "multicast_server", "socket_opts",
+                       "tcp_server", "timeout_server", "udp_server",
+                       "multicast_client", "send_recv", "tcp_client",
+                       "timeout_client",  "udp_client" ]
+        wasm_apps_dir = args.working_directory
+        print("Compile wasm app into aot files")
+        for wasm_file in wasm_files:
+            CMD = []
+            CMD.append(WAMRC_CMD)
```

Comment:
[minor] not sure what's the benefit of using `append` here if the list can be constructed in place, i.e.
```python
CMD = [
	WAMRC_CMD,
	'-o',
	# ...
```

Also, I think upper case variable names are used for global variables, so consider renaming it to `cmd`